### ISSUE TITLE
Error occurred during asynchronous operation processing

### ISSUE THREAD

**Issue by marlon-shiftone** [keras-team/keras#14597]
I' training a model CNN using data generator. 
```
def data_generator(file_list, batch_size=5):
    i = 0
    while True:        
        if i >= len(file_list):  # This loop is used to run the generator indefinitely.
            i = 0
            #np.random.shuffle(file_list)
        else:
            list_df = split_chunks(file_list, chunk=200)
            for i in range(len(list_df)):
               X, y = make_hypercubes(list_df[i], dim_4=96, dim_3=100)
               X = np.array(X)
               y = np.array(y)
               X = X.reshape(-1, 96, 10, 10, list_df[i][:, 0:-1].shape[-1])            
               yield X, y
            del list_df   
            gc.collect()
            i = i + 1    

```

In order to do this I have to slice the data and reshape it. However, at `steps_per_epoch` 720 of `epoch` 1 it gives me the error:

    async_operation_processor.py:138] Error occurred during asynchronous operation processing: Cannot log infinite or NaN value to attribute metrics/batch

I have already printed the `y_pred` and `y_true` inside the loss and I can see that it gives some empty lists `[]` which explode the gradient. So I've tried to check if my slices are producing the empty array with:

   ```

    from sys import exit
    for i in X_seq:
        if not i.any():
            print("Error with X.")
            exit(0)
    for i in y_seq:
        if not i.any():
            print("Error with Y.")
            exit(0)
```

But apparently, it is not since it doesn't print and stops in any moment. So it seems the problem is not with the input. Any idea?

Thanks!
